ecms_base/modules/custom/ecms_icon_library/src/Plugin/Field/FieldFormatter/EcmsIconLibraryFormatter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_icon_library\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\media\Entity\Media;
+use Drupal\file\Entity\File;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Field\FormatterBase;
+
+/**
+ * Plugin implementation of the 'image' formatter.
+ *
+ * We have to fully override standard field formatter, so we will keep original
+ * label and formatter ID.
+ *
+ * @FieldFormatter(
+ *   id = "ecms_icon_library_formatter",
+ *   label = @Translation("eCMS Icon Formatter"),
+ *   field_types = {
+ *     "ecms_icon_library"
+ *   },
+ * )
+ */
+class EcmsIconLibraryFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
```

Comment:
add return type - might look like "? array" or you might be able to use "array" and return "[]" when an empty return is needed.